corehq/apps/export/static/export/js/create_export.js

```diff
@@ -0,0 +1,311 @@
+        });
+
+        // Behavior related to selecting an app type and an app
+        self.hasSpecialAppTypes = ko.observable(false);
+        self.hasNoCaseTypes = ko.observable(false);
+        self.showAppType = ko.computed(function () {
+            return self.hasSpecialAppTypes() || self.isFormModel();
+        });
+
+        self.showHasNoCaseTypes = ko.computed(function () {
+            return self.hasNoCaseTypes() && self.application();
+        });
+
+        // Behavior related to selecting a form
+        self.showMislabeled = ko.observable(false);
+        self.showSuggestions = ko.observable(false);
+        self.showAppDoesNotExist = ko.observable(false);
+        self.showDuplicate = ko.observable(false);
+        self.duplicatePossibilities = ko.observableArray();
+
+        self.showSubmissionCount = ko.computed(function () {
+            return self.modelType() === 'form' && !_.isEmpty(self.selectedFormData());
+        });
+        self.submissionCountText = ko.observable('');
+
+        self.selectedFormData.subscribe(function (newValue) {
+            var text = '';
+            if (self.modelType() === 'form') {
+                if (newValue.submissions === 1) {
+                    text = gettext('<%= count %> form submission available.');
+                } else {
+                    text = gettext('<%= count %> form submissions available.');
+                }
+                text = _.template(text)({ count: newValue.submissions });
+            }
+            self.submissionCountText(text);
+
+            self.showMislabeled(self.appType() === '_unknown' && !self.form() && !newValue.app_copy);
+            self.showSuggestions(self.appType() === '_unknown' && !self.form() && !newValue.no_suggestions);
+            self.showAppDoesNotExist(newValue.app_does_not_exist);
+            self.showDuplicate(newValue.duplicate);
+            self.duplicatePossibilities(newValue.possibilities);
+            self.selectedFormName(newValue.is_user_registration ? gettext('User Registration') : newValue.name);
+        });
+
+        // Behavior related to initializing the drilldown form
+        var drilldownDefaults = {
+            model_type: '',
+            app_type: 'all',
+            application: null,
+            module: null,
+            form: null,
+            case_type: null,
+        };
+
+        self.setAppTypes = function () {
+            self.appType(drilldownDefaults.app_type);
+
+            var $formElem = $('#id_app_type');
+            if ($formElem.length > 0) {
+                $formElem.select2({
+                    data: self._app_types || [],
+                    triggerChange: true,
+                }).select2('val', drilldownDefaults.app_type).trigger('change');
+            }
+        },
+        self._initSelect2 = function (observable, fieldSlug) {
+            return function (fieldData) {
+                if (fieldSlug) {
+                    observable('');
```

Comment:
What is this?

Reply:
It clears the value of the field. I'm not 100% sure it's necessary, it's just the equivalent of [this line](https://github.com/dimagi/commcare-hq/pull/22128/files/4594a7c493720fc970a6da53be57b4c3e0bff56f#diff-70abf7231c07d7517f2229435dda567fL88).

Reply:
Interesting.  I just can't imagine the new line having any side-effects (but I've been wrong before...)